zulipterminal/ui_tools/boxes.py

```diff
@@ -940,54 +940,59 @@ def soup2markup(cls, soup: Any, metadata: Dict[str, Any],
         }
         unrendered_template = '[{} NOT RENDERED]'
         for element in soup:
+            if isinstance(element, Tag):
```

Comment:
I really like the simplification this enables, though my only thought would be how this isn't part of the if/elif chain. Use of the `tag` prefix names elsewhere imply that this is how it works, but we might consider a comment to the effect of we're caching/naming entries used in all but the first element of the chain to improve legibility/lookup, and a blank line between this conditional and the main if/elif chain.

Reply:
Yes I was considering adding a comment. This would certainly help first-timers :+1: